core/base/discreteGradient/DiscreteGradient.cpp

```diff
@@ -24,12 +24,31 @@ void DiscreteGradient::initMemory(const AbstractTriangulation &triangulation) {
   }
 
   // clear & init gradient memory
-  for(int i = 0; i < dimensionality_; ++i) {
-    (*gradient_)[2 * i].clear();
-    (*gradient_)[2 * i].resize(numberOfCells[i], -1);
-    (*gradient_)[2 * i + 1].clear();
-    (*gradient_)[2 * i + 1].resize(numberOfCells[i + 1], -1);
+#ifdef TTK_ENABLE_OPENMP
+#pragma omp parallel master num_threads(threadNumber_)
```

Comment:
assuming we're dealing with a 3D dataset (i = 0, i = 1, i = 2, i = 3), does it mean that the code will now use 8 threads? (or 2 threads 4 times, one dimension at a time; in other words, how will the task handle the for loop?)

Reply:
The code will use `threadNumber_` threads. The master thread will create the tasks for other threads to compute. It is the only thread that will go through the `for` loop. The other threads will only compute the tasks from the task pool. They may be not enough tasks to have one task for one thread, in which case some thread may stay idle.